test/support/cluster/orchestrator.rb

```diff
@@ -157,27 +156,35 @@ def save_config(clients)
     clients.each { |c| c.cluster(:saveconfig) }
   end
 
-  def wait_cluster_building(clients, max_attempts: 200)
-    attempt_count = 0
+  def wait_cluster_building(clients, max_attempts: 600)
+    wait_for('cluster building', clients, max_attempts) do |client|
+      info = hashify_cluster_info(client)
+      info['cluster_state'] == 'ok'
+    end
+  end
 
-    clients.each do |client|
-      loop do
-        info = hashify_cluster_info(client)
-        attempt_count += 1
-        break if info['cluster_state'] == 'ok' || attempt_count > max_attempts
-        sleep 0.1
-      end
+  def wait_failover(master_key, slave_key, clients, max_attempts: 600)
+    wait_for('cluster failover', clients, max_attempts) do |client|
+      flags = hashify_cluster_node_flags(client)
+      flags[master_key] == 'slave' && flags[slave_key] == 'master'
     end
   end
 
-  def wait_failover(master_key, slave_key, clients, max_attempts: 200)
-    attempt_count = 0
+  def wait_replication_delay(clients, timeout_sec)
+    timeout_msec = timeout_sec.to_i * 1000
+    wait_for('replication delay', clients, clients.size + 1) do |client|
+      client.wait(1, timeout_msec) if client.role.first == 'master'
+      true
+    end
+  end
 
+  def wait_for(state, clients, max_attempts)
+    attempt_count = 1
     clients.each do |client|
-      loop do
-        flags = hashify_cluster_node_flags(client)
+      attempt_count.step(max_attempts) do |i|
+        p "Reached max attempts at waiting for #{state}" if i >= max_attempts
```

Comment:
Is this print a forgotten debug statement?

Reply:
No, it isn't. It is on purpose. There's a possibility that test failure is occured when reaching max attempts. I think that the print is better than nothing.

Is this print redundant or noisy?

Reply:
I'm sorry, I misunderstood. I removed the statement and added `break` step.